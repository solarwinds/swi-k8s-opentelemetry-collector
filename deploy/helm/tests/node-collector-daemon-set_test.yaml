# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test for node-collector-daemon-set
templates:
  - node-collector-daemon-set.yaml
  - node-collector-config-map.yaml
  - common-env-config-map.yaml
  - network/configmap.yaml
tests:
  - it: DaemonSet spec should match snapshot when using default values
    template: node-collector-daemon-set.yaml
    asserts:
      - matchSnapshot:
          path: spec.template.spec
  - it: DaemonSet spec should match snapshot when ebpfNetworkMonitoring is enabled
    template: node-collector-daemon-set.yaml
    set:
      ebpfNetworkMonitoring.enabled: true
    asserts:
      - matchSnapshot:
          path: spec.template.spec