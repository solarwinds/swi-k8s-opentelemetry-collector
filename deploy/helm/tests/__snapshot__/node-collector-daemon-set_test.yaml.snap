DaemonSet spec should match snapshot when ebpfNetworkMonitoring is enabled:
  1: |
    affinity:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
            - matchExpressions:
                - key: eks.amazonaws.com/compute-type
                  operator: NotIn
                  values:
                    - fargate
    containers:
      - command:
          - /wrapper
          - /swi-otelcol
          - --config=/conf/relay.yaml
        env:
          - name: CHECKPOINT_DIR
            value: /var/lib/swo/checkpoints
          - name: POD_NAME
            valueFrom:
              fieldRef:
                fieldPath: metadata.name
          - name: POD_NAMESPACE
            valueFrom:
              fieldRef:
                fieldPath: metadata.namespace
          - name: NODE_NAME
            valueFrom:
              fieldRef:
                fieldPath: spec.nodeName
          - name: SOLARWINDS_API_TOKEN
            valueFrom:
              secretKeyRef:
                key: SOLARWINDS_API_TOKEN
                name: solarwinds-api-token
                optional: true
        envFrom:
          - configMapRef:
              name: RELEASE-NAME-swo-k8s-collector-common-env
        image: solarwinds/swi-opentelemetry-collector:0.8.8
        imagePullPolicy: IfNotPresent
        livenessProbe:
          httpGet:
            path: /
            port: 13133
        name: swi-opentelemetry-collector
        ports:
          - containerPort: 8888
            name: http
            protocol: TCP
        readinessProbe:
          httpGet:
            path: /
            port: 13133
        resources:
          limits:
            memory: 700Mi
          requests:
            memory: 50Mi
        volumeMounts:
          - mountPath: /var/log/pods
            name: varlogpods
            readOnly: true
          - mountPath: /var/log/containers
            name: varlogcontainers
            readOnly: true
          - mountPath: /var/lib/docker/containers
            name: varlibdockercontainers
            readOnly: true
          - mountPath: /conf
            name: opentelemetry-collector-configmap
            readOnly: true
          - mountPath: /run/log/journal
            name: runlogjournal
            readOnly: true
          - mountPath: /var/log/journal
            name: varlogjournal
            readOnly: true
          - mountPath: /var/lib/swo/checkpoints
            name: logcheckpoints
    nodeSelector:
      kubernetes.io/arch: amd64
      kubernetes.io/os: linux
    securityContext:
      fsGroup: 0
      runAsGroup: 0
      runAsUser: 0
    terminationGracePeriodSeconds: 30
    tolerations:
      - effect: NoSchedule
        operator: Exists
    volumes:
      - hostPath:
          path: /var/log/pods
        name: varlogpods
      - hostPath:
          path: /var/log/containers
        name: varlogcontainers
      - hostPath:
          path: /var/lib/docker/containers
        name: varlibdockercontainers
      - hostPath:
          path: /run/log/journal
        name: runlogjournal
      - hostPath:
          path: /var/log/journal
        name: varlogjournal
      - hostPath:
          path: /var/lib/swo/checkpoints/<CLUSTER_UID>
          type: DirectoryOrCreate
        name: logcheckpoints
      - configMap:
          items:
            - key: logs.config
              path: relay.yaml
          name: RELEASE-NAME-swo-k8s-collector-node-collector-config
        name: opentelemetry-collector-configmap
DaemonSet spec should match snapshot when using default values:
  1: |
    affinity:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
            - matchExpressions:
                - key: eks.amazonaws.com/compute-type
                  operator: NotIn
                  values:
                    - fargate
    containers:
      - command:
          - /wrapper
          - /swi-otelcol
          - --config=/conf/relay.yaml
        env:
          - name: CHECKPOINT_DIR
            value: /var/lib/swo/checkpoints
          - name: POD_NAME
            valueFrom:
              fieldRef:
                fieldPath: metadata.name
          - name: POD_NAMESPACE
            valueFrom:
              fieldRef:
                fieldPath: metadata.namespace
          - name: NODE_NAME
            valueFrom:
              fieldRef:
                fieldPath: spec.nodeName
          - name: SOLARWINDS_API_TOKEN
            valueFrom:
              secretKeyRef:
                key: SOLARWINDS_API_TOKEN
                name: solarwinds-api-token
                optional: true
        envFrom:
          - configMapRef:
              name: RELEASE-NAME-swo-k8s-collector-common-env
        image: solarwinds/swi-opentelemetry-collector:0.8.8
        imagePullPolicy: IfNotPresent
        livenessProbe:
          httpGet:
            path: /
            port: 13133
        name: swi-opentelemetry-collector
        ports:
          - containerPort: 8888
            name: http
            protocol: TCP
        readinessProbe:
          httpGet:
            path: /
            port: 13133
        resources:
          limits:
            memory: 700Mi
          requests:
            memory: 50Mi
        volumeMounts:
          - mountPath: /var/log/pods
            name: varlogpods
            readOnly: true
          - mountPath: /var/log/containers
            name: varlogcontainers
            readOnly: true
          - mountPath: /var/lib/docker/containers
            name: varlibdockercontainers
            readOnly: true
          - mountPath: /conf
            name: opentelemetry-collector-configmap
            readOnly: true
          - mountPath: /run/log/journal
            name: runlogjournal
            readOnly: true
          - mountPath: /var/log/journal
            name: varlogjournal
            readOnly: true
          - mountPath: /var/lib/swo/checkpoints
            name: logcheckpoints
    nodeSelector:
      kubernetes.io/arch: amd64
      kubernetes.io/os: linux
    securityContext:
      fsGroup: 0
      runAsGroup: 0
      runAsUser: 0
    terminationGracePeriodSeconds: 30
    tolerations:
      - effect: NoSchedule
        operator: Exists
    volumes:
      - hostPath:
          path: /var/log/pods
        name: varlogpods
      - hostPath:
          path: /var/log/containers
        name: varlogcontainers
      - hostPath:
          path: /var/lib/docker/containers
        name: varlibdockercontainers
      - hostPath:
          path: /run/log/journal
        name: runlogjournal
      - hostPath:
          path: /var/log/journal
        name: varlogjournal
      - hostPath:
          path: /var/lib/swo/checkpoints/<CLUSTER_UID>
          type: DirectoryOrCreate
        name: logcheckpoints
      - configMap:
          items:
            - key: logs.config
              path: relay.yaml
          name: RELEASE-NAME-swo-k8s-collector-node-collector-config
        name: opentelemetry-collector-configmap
